Unified Dashboard: Remove tabs and update quick links layout

ravishanker opened the pull request:
Part of https://github.com/wordpress-mobile/WordPress-Android/issues/19270
Fixes #. See pbArwn-6dL-p2#comment-7813

## Description
This PR updates the following based on above
- Removes tabs (Home | Menu) on My Site for Jetpack app 
- Updates layout of Quick Links menu from pills in horizontal scroll to vertical items inside a card
- Adds a personalization screen for quick links - Users can add or remove any quick links from the dashboard using personalization screen


📔 No changes to WP app dashboard

## Jetpack app - Before and After 

| Home | Menu | Home | Menu |
|--------|--------|--------|--------|
| ![Screenshot_20230907_120535](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/68d52f83-12df-4c48-a052-814dc45d1519) |![Screenshot_20230907_120553](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/f1a039b8-f85e-4839-a3c1-14e5a3b1bc64) | ![Screenshot_20230908_145232](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/64d927b5-1ab0-4221-b98e-2d15cdfb0716) | ![Screenshot_20230908_165212](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/3a23e884-7349-4ce9-8b88-a375cbeac122) |

## Personalization screen for QuickLinks
<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/3aa17767-ba0c-4ee9-b448-3058457384e4" width="210" height="480">

## Known issues 
- The Snackbar prompt for quick start is not working as expected for tasks which are part of quick links. Since any quick links can be removed, the quick start focus points for Pages, Stats, Media and Social sharing is shown inside More menu. Although the focus points are shown correctly, the prompt for the next task is not shown correctly. 
- The personalization screen tabs are not swipeable, this will be taken up in a seperate PR cc: @zwarm 

## To test:

### **Jetpack app**
#### Dashboard is working as expected
- Launch JP app
- `Verify` the Home now looks as in third screen shot above
-  `Tap` on Quick links and verify that it is working as expected
-   Go to `Me` → `App Settings` → `Privacy Settings` → `Enable Collect Information`
-  `Verify` that the tracking for quick link is working as expected

#### More menu is working as expected 
-  Go to dashboard 
- `Tap` on ... More
- `Verify` it goes to My Site menu as in fourth screen shot above
- `Tap` on each of the menu items and `Verify` they all work as before
- `Verify` that the tracking for quick link is working as expected
- `Test` **Quick Start** flow
- `Verify` that the app goes through the flow as before

#### Personalization screen is working as expected
- Go to dashboard 
- `Tap` on Personalize your home tab
-  Go to `Shortcuts` tab
- Hide/Show any quick links 
- Verify that the same is reflected on the dashboard

#### Signup flow 
- Sign up on the Jetpack app with an email
- Click on the magic link
- Verify that the login works as expected
- Verify that no sites view is shown as before

### **WordPress app**
- Launch WP app
- `Verify` it only shows menu on My Site as before 


## Regression Notes
1. Potential unintended areas of impact
Home

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Existing unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

ravishanker committed:
Update my_site_fragment.xml

- Remove tabs
- Remove ViewPager
- Add RecyclerView for MySiteItems

ravishanker committed:
Update MySiteFragment.kt

Update MySiteItems on MySiteFragment, that was previously in MySiteTabFragment

ravishanker committed:
Update quick links layout

- Update layout to rows in a card

ravishanker committed:
Update QuickLinkRibbonBuilder.kt

- Add More item

ravishanker committed:
Update navigation

- Update navigation to handle More item on quick links

ravishanker committed:
Update AnalyticsTracker.java

ravishanker committed:
Update AnalyticsTrackerNosara.java

ravishanker committed:
Create MySiteMenuActivity

- Create MySite Menu activity
- Reuse existing MySiteTabFragment

ravishanker committed:
Update MySiteTabFrament

Update MySiteTabFragment for reuse for More navigation to Menu

ravishanker committed:
hook up Menu to More button

ravishanker committed:
Update MySiteFragment.kt

- Fix Detekt reported issues

ravishanker committed:
remove unused dimens

ravishanker committed:
Update CardsBuilderTest.kt

ravishanker committed:
Update QuickLinkRibbonBuilderTest.kt

ravishanker committed:
Update CardsBuilderTest.kt

remove unused import

ravishanker commented:
@pachlava - FYI, Making changes to dashboard in the PR.  Need your help in looking into UI tests. 🙇

AjeshRPai reviewed (changes requested):
Hey @ravishanker 

I took the build and smoke tested the app. Below are my findings 

1. We need to remove the Initial screen setting and set the default value to Dashboard, otherwise the Quick start card would show up in the menu screen. Any quick start task that has a quick start focus - for eg: **Connect with Other sites**  in the dashboard will crash on click

<img width=320 src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/2d601d03-e858-4bc6-8cda-e9f4aab3b3f9" />

[quick_start_settings.webm](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/d5a4c08d-f8d5-4263-812d-2c7dab700b93)

2. Also I feel that we should remove the Jetpack migration card on menu(In the Jetpack App). We are already showing it in Dashboard, I think its irrelevant to show in menu 🤔

osullivanchris commented:
@ravishanker  hey thanks for pulling this together. It feels good on the whole. I've put together some specs to tighten things up. 

First are some header tweaks we can make to align nicer with the cards (see the alignments in blue lines)
- add 8dp left padding to the site icon (currently none)
- change site icon to 40dp (currently 56) and radius 5 (to look nice with cards 10)
- 8dp padding between site icon and site title (currently 16)
- 12dp right padding on the chevron at the end of the row (currently none)

In the new list itself, some other minor visual changes:
- can we make the icons black per the deeper page
- change the spacing between the icon and the list item name to 16dp (currently 8). Aligns the site title and list items both with the 72dp keyline sometimes advised
- change list item weight to medium to align with 16dp medium used throughout cards below. Site title can be same size/weight or retain dynamic size we have now.

ellipsis icon:
- And one for you and @AjeshRPai . could we use the new ellipsis icon below to match the other new icons used. Can we use for both the More list item, and the cards? 
- one for @AjeshRPai  - can we change the right padding for the ellipsis to be 8dp rather than 16dp?
- 
<img width="1231" alt="Android quick actions specs" src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/8527029a-fce6-4eeb-8fac-168388a0a80b">


Referencing @AjeshRPai questions above:
- We removed the choice of dashboard / menu on iOS
- I agree we should no longer show any cards on the menu page

pachlava commented:
> @pachlava - FYI, Making changes to dashboard in the PR. Need your help in looking into UI tests. 🙇

Thanks for the ping @ravishanker 👋 

The reason why two dashboard cards UI tests fail is because Espresso can't scroll to cards in current PR:

```
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match 
one or more of the following constraints:
(view has effective visibility <VISIBLE> 
and is descendant of a view matching (is assignable from class <class android.widget.ScrollView> 
or is assignable from class <class android.widget.HorizontalScrollView> 
or is assignable from class <class androidx.core.widget.NestedScrollView> 
or is assignable from class <class android.widget.ListView>))
```

While I'm not entirely sure that this is the exact reason, I can see that top `NestedScrollView` has been replaced with `CoordinatorLayout` in `my_site_tab_fragment.xml`.

With my limited knowledge around this topic, I tried playing around locally with adding `ScrollView` or `NestedScrollView` back into this file, and also into `my_site_dashboard_cards.xml`. I obtained the best results when I added `ScrollView` as a root item into `my_site_dashboard_cards.xml`: after that, the native Espresso `scrollTo()` started working again, the `e2ePagesCardNavigation` started passing, but `e2eActivityLogCardNavigation` still failed, but because of a different reason: "Activity Log" card was not properly shown on screen any longer, most probably because I messed up the elements sizing / loading when I added the tag.

Because of this, I made no commits, but what I described above is at least one way to solve it, if done properly (which you have much higher chances to do than me). Please let me know if you need any more help 🙂

AjeshRPai committed:
Merge branch 'issue/18956-dashboard-personalization-handle-no-cards-in-dashboard' into UI-Modernization-Remove-tabs-and-update-quick-links-layout

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt

AjeshRPai committed:
+ Adds: the logic for blogging prompt card navigation

AjeshRPai committed:
Merge branch 'trunk' into UI-Modernization-Remove-tabs-and-update-quick-links-layout

AjeshRPai committed:
- Removes: Jetpack security item

AjeshRPai committed:
[WIP] Handle More icon click

zwarm committed:
Merge branch 'trunk' into UI-Modernization-Remove-tabs-and-update-quick-links-layout

zwarm committed:
Refactor: remove constraints for scrolling on dashboard

zwarm committed:
Remove goToMenuTab because the dashboard no longer contains home and menu tabs.

zwarm committed:
[WIP] Commenting out e2eAllDayStatsLoad. Will return to this

zwarm committed:
Refactor: Remove onCreateSiteResult() - tabs are no longer used

zwarm committed:
Refactor: Rename my_site_tab_fragment to my_site_menu_fragment

zwarm committed:
Refactor: Rename MySiteTabFragment to MySiteMenuFragment

zwarm committed:
Refactor: Remove MySiteTabsAdapter

zwarm committed:
Refactor: Move MySiteMenuFragment out of tabs package.

zwarm committed:
Remove call to viewModel.onCreateSiteResult

zwarm committed:
Remove initial screen for app settings view

zwarm committed:
Add UnifiedMySiteMenuActivity entry

zwarm committed:
Add openUnifiedMySiteMenu navigation action to launch UnifiedMySiteMenuActivity

zwarm committed:
Add quick start focus into its own layout file so it can be used within the unified my site menu

zwarm committed:
[WIP] Route OpenMore to UnifiedMySiteMenu instead of MySiteMenuActivity

zwarm committed:
[WIP] Initial implementation of the unified my site menu activity

zwarm committed:
[WIP] Initial implementation of the unified my site menu view model

zwarm committed:
Refactor: rename view model to include MySite to match the activity

zwarm committed:
[WIP] First styling pass

zwarm committed:
Refactor: Use spacer for empty category header

zwarm committed:
Remove some log lines and comments

zwarm committed:
Remove unused import statement

zwarm committed:
Fix detekt long line:

zwarm committed:
Refactor: Rename the UnifiedMySiteMenu classes to drop the Unified and MySite words and move to a separate package under MySite

zwarm committed:
Remove test related to default tab pref

AjeshRPai committed:
- Removes: the unified naming from compose views

AjeshRPai committed:
Merge pull request #19244 from wordpress-mobile/issue/unified-dashboard-refactoring

Unified Dashboard: Refactoring Part 1

AjeshRPai committed:
↑ Updates: the layout to make it scrollable always

AjeshRPai committed:
- Removes: the redundant relative layout

AjeshRPai committed:
- Removes: tab ui step logic and live data as the tabs have removed

AjeshRPai committed:
- Removes: the redundant SiteInfoToolbarViewParams class

AjeshRPai committed:
- Removes: the toolbar scrolling logic as its no longer needed

AjeshRPai committed:
- Removes: tab related logic from QuickStartRepository

AjeshRPai committed:
- Removes: the default tab and tab tracking logic from the viewmodel

AjeshRPai committed:
* Fixes: Scroll to site header on active quick start

AjeshRPai committed:
- Removes: on track with tab source logic from viewmodel

AjeshRPai committed:
- Removes: tracking of tab in quick start

AjeshRPai committed:
- Removes: the default tab logic from my site viewmodel

AjeshRPai committed:
- Removes: default tab logic to show focus points from builder

AjeshRPai committed:
↑ Updates: the visibility logic of app bar when site is present

AjeshRPai committed:
- Removes: cardAndItems from state and updates logic of cards & items

AjeshRPai committed:
- Removes: site info header state

- Site info header state was introduced to fix a flickering issue in
the tab layout + collapsible toolbar

AjeshRPai committed:
- Removes: redudndat tab source variable

AjeshRPai committed:
* Fixes: lint issues

AjeshRPai committed:
- Removes: tab variable used for tracking

AjeshRPai committed:
* Fixes: quick link ribbon tests

AjeshRPai committed:
* Fixes: Quick start repository tests

AjeshRPai committed:
* Fixes: SiteInfoHeaderCardViewModelSliceTest

AjeshRPai committed:
* Fixes: SiteItemsViewModelSliceTest

AjeshRPai committed:
* Fixes: MySiteViewModelTest

AjeshRPai committed:
* Fixes: tests in MySiteViewModelTest

AjeshRPai committed:
* Fixes: jetpack badge scenario

AjeshRPai committed:
- Removes: the fetch for dashboard items in tests

AjeshRPai committed:
- Removes: MySiteMenuActivity, MySiteMenuFragment

zwarm committed:
Refactor: Move the interface out of MySiteTabFragment into its own file

zwarm committed:
Remove references to MySiteMenu activity and fragment

zwarm committed:
Remove unused references

zwarm committed:
Remove test for backup menu item visibility; it is no longer listed on the dashboard

zwarm committed:
Remove tests for backup and scan menu item visibility; they are no longer listed as individual items on the dashboard

zwarm committed:
Revert "Remove test for backup menu item visibility; it is no longer listed on the dashboard"

This reverts commit 8ac8b81bfb3de338beeca21b2b1df5fc8ef9103f.

zwarm committed:
Revert "Remove tests for backup and scan menu item visibility; they are no longer listed as individual items on the dashboard"

This reverts commit a1aa2a18fec9abbbb050e012a7d154cd1028a1cc.

zwarm committed:
[WIP] Comment out unit tests for the moment, so refactoring can continue

zwarm committed:
[WIP] Comment unused test helper functions

AjeshRPai committed:
- Removes: redundant test as the focus point will always be false

AjeshRPai committed:
- Removes: Unnecessary mockito stubbings

AjeshRPai committed:
- Removes: Redundant imports

AjeshRPai committed:
* Disables: the e2e tests related to dashboard temporarily

AjeshRPai committed:
+ Adds: string resources for tab titles in personalization screen

AjeshRPai committed:
+ Adds: string resources for tab titles in personalization screen

AjeshRPai committed:
+ Adds: tabs in personalization screen

AjeshRPai committed:
+ Adds: tabs in personalization screen

AjeshRPai committed:
↑ Updates: the string resource naming to include tabs

AjeshRPai committed:
↑ Updates: the string resource naming to include tabs

AjeshRPai committed:
+ Adds: PersonalizationShortcutsViewModelSlice and ShortcutsState

AjeshRPai committed:
+ Adds: PersonalizationShortcutsViewModelSlice and ShortcutsState

AjeshRPai committed:
+ Adds: viewmodel slice for shortcuts personalization

AjeshRPai committed:
+ Adds: viewmodel slice for shortcuts personalization

AjeshRPai committed:
↑ Updates: the logic to fetch shortcuts

AjeshRPai committed:
↑ Updates: the logic to fetch shortcuts

AjeshRPai committed:
+ Adds: the logic to show shortcuts state in tab

AjeshRPai committed:
+ Adds: the logic to show shortcuts state in tab

AjeshRPai committed:
↑ Updates: the ui for shortcuts personalization tab

AjeshRPai committed:
↑ Updates: the ui for shortcuts personalization tab

AjeshRPai committed:
↑ Updates: the ui for shortcuts icon

AjeshRPai committed:
↑ Updates: the ui for shortcuts icon

AjeshRPai committed:
* Renames:PersonalizationShortcutsViewModelSlice to

* Renames:PersonalizationShortcutsViewModelSlice to
ShortcutsPersonalizationViewModelSlice

AjeshRPai committed:
* Renames:PersonalizationShortcutsViewModelSlice to

* Renames:PersonalizationShortcutsViewModelSlice to
ShortcutsPersonalizationViewModelSlice

AjeshRPai committed:
→ Moves: Dashboard card personalization to a viewmodel slice

AjeshRPai committed:
→ Moves: Dashboard card personalization to a viewmodel slice

AjeshRPai committed:
* Renames: PersonalizationViewModelTest

* Renames: PersonalizationViewModelTest
to DashboardCardPersonalizationViewModelSliceTest

AjeshRPai committed:
* Renames: PersonalizationViewModelTest

* Renames: PersonalizationViewModelTest
to DashboardCardPersonalizationViewModelSliceTest

AjeshRPai committed:
+ Adds: Image resources for action button in shortcuts

AjeshRPai committed:
+ Adds: Image resources for action button in shortcuts

zwarm committed:
Merge pull request #19263 from wordpress-mobile/issue/unified-dashboard-refactoring-part-2-site-header

Unified Dashboard: Refactoring Part 2 - Remove Tabs from viewmodel and view

AjeshRPai committed:
+ Adds: Icon button for adding/removing shortcuts

AjeshRPai committed:
+ Adds: Icon button for adding/removing shortcuts

AjeshRPai committed:
Merge remote-tracking branch 'origin/issue/unified-dashboard-refactoring-implement-personalization-quick-links' into issue/unified-dashboard-refactoring-implement-personalization-quick-links

AjeshRPai committed:
* Fixes: Icon tint color on action button

zwarm committed:
Merge pull request #19271 from wordpress-mobile/issue/unified-dashboard-refactoring-implement-personalization-quick-links

UI - Implement personalization screen for quick links

zwarm committed:
Resolve next steps conflicts

zwarm committed:
Remove NEXT_STEPS; it is no longer part of the view

zwarm committed:
Add ListItemAction to ListItem so the item type can be identified in shortcuts

zwarm committed:
Assign ListItemAction when building a listItem

zwarm committed:
Assign listItemAction is previews

AjeshRPai committed:
+ Adds: the logic to keep the show/hide of quick link in preferences

AjeshRPai committed:
+ Adds: the logic to keep the show/hide of default quick link

AjeshRPai committed:
+ Adds: the wrapper for show/hide of default quick link and list items

AjeshRPai committed:
* Fixes: default quick link and site item fetch

AjeshRPai committed:
+ Implements: the logic to fetch whether the shortcuts is active or not

AjeshRPai committed:
+ Adds: the logic to seperate active and inactive shortcuts

AjeshRPai committed:
+ Adds: the logic to hide active/inactive shortcuts if empty

AjeshRPai committed:
+ Adds: the logic of removing and adding shortcuts on vm

+ Adds: ListItemAction attribute to the state so that we can get which
shortcut is clicked
+ Adds: the logic to set the preference to hide/show

AjeshRPai committed:
↑ Updates: the parameter naming for readability

AjeshRPai committed:
+ Adds: the logic to hide active/inactive shortcuts in parent viewmodel

AjeshRPai committed:
+ Adds: the logic for handling the action button click

AjeshRPai committed:
+ Adds: a ListItemActionHandler to handle the click on list item

AjeshRPai committed:
+ Adds: List action handler to handle the site items click

AjeshRPai committed:
+ Adds: QuickLinksItemViewModelSlice

+ Adds: QuickLinksItemViewModelSlice to show the logic of active
quick links

AjeshRPai committed:
- Removes: redundantQuickLinkRibbonBuilder.kt

zwarm committed:
Add listItemAction to all fixtures

AjeshRPai committed:
- Removes: QuickLinkRibbonBuilder from card builder

zwarm committed:
Add ListItemAction when during init of ListItem

AjeshRPai committed:
↑ Updates: the viewholder to remove stats,pages and media focus points

AjeshRPai committed:
↑ Updates: the SiteItemsViewModelSlice to use list action handler

AjeshRPai committed:
↑ Updates: the MenuViewModel to use list action handler

AjeshRPai committed:
↑ Updates: the quick link ribbon to remove pages,stats and media QS

AjeshRPai committed:
+ Adds: QuickLinkRibbonViewModelSlice to MySiteViewModel

+ Adds: QuickLinkRibbonViewModelSlice to MySiteViewModel

AjeshRPai committed:
* Fixes: tests in CardBuilderTest

zwarm committed:
Merge pull request #19273 from wordpress-mobile/issue/19270-unified-menu-refactorings

Dashboard: unified menu Refactoring

AjeshRPai committed:
* Fixes: Detekt issues

AjeshRPai committed:
* Fixes: lint issues

AjeshRPai committed:
- Removes: quick link ribbon builder test

AjeshRPai committed:
- Removes: quick link ribbon builder tests and dependencies

AjeshRPai committed:
Merge branch 'UI-Modernization-Remove-tabs-and-update-quick-links-layout' into issue/19270-quick-link-ribbon-builder

# Conflicts:
#	WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteCardAndItemTest.kt

AjeshRPai committed:
→ Moves: the tests of site navigation to ListItemActionHandlerTest

zwarm committed:
Fix detekt issue

zwarm committed:
Refactor: replace git add . with a let to fix a crash on login with a user that has no sites

zwarm committed:
Replace double bang with an explicity let check so to avoid crash when in a no sites situation

zwarm committed:
Refactor: replace requireNotNull with a let check; because quickLinksItems can be null when we are in a no-site situation

zwarm committed:
Refactor: update tests to support quickLinksViewModelSlice, remove quick link navigation tests, and general cleanup of unnecc stubbings.

zwarm committed:
Merge pull request #19279 from wordpress-mobile/issue/19270-quick-link-ribbon-builder

Implements: the quick link item logic in Unified Menu

AjeshRPai committed:
- Removes: QuickStartExistingUsersV2FeatureConfig.kt

AjeshRPai committed:
- Removes: QuickStartExistingUsersV2FeatureConfig from tests

AjeshRPai committed:
- Removes: MySiteDashboardTabsFeatureConfig.kt from tests and gradle

AjeshRPai committed:
+ Adds: logic to set the quick start item in quick link items

AjeshRPai committed:
+ Adds: the list item action as an identifier for quick start check

AjeshRPai committed:
+ Adds: the logic to show quick start focus point in quick links

AjeshRPai committed:
- Removes: MySiteTabsFeatureFlag from build wrappper

AjeshRPai committed:
↑ Updates: the padding and spacing in Quick links

AjeshRPai committed:
+ Adds: disable tint attribute

AjeshRPai committed:
* Fixes: RTL for quick links

AjeshRPai committed:
+ Adds: Quick link build logic when there is a site change

AjeshRPai committed:
+ Adds: Quick link build logic when there is a site change

AjeshRPai committed:
+ Adds: the logic for OnResume refresh of MySiteViewModel

AjeshRPai committed:
↑ Updates: the naming of quick link ribbon to item

osullivanchris commented:
@AjeshRPai nice work this is looking really good and working well. a few small visual issues:

1 - when I go into the full menu there's something wrong with the text styles. They have changed. They look heavier and grey rather than regular black text from before. And also the space between the icon and text looks tighter. The text style in the personalisation>shortcuts tab seems the same. It can also change to the way the menu was before.


<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/5192fa2b-0c9b-4e31-aa7c-f25e53976aad" width="250">

2 - the color of the separators between the shortcuts should be the same as the card border around it, but looks slightly different right now. I also removed all cards, then added others back, and the separator color changed to a darker color 

<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/4b985c92-fac4-403e-97f0-aff2edc1fc12" width="250">

<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/f625b956-220e-454e-ad84-2bd748cdf89b" width="250">

3 - Can you add some extra space above the 'Active shortcuts' and 'Inactive shortcuts' headings? It all looks a bit jammed up together.

<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/a51374a5-1f0c-4400-ac2e-46a8970f5a7f" width="250">

4 - the style and label 'more' looks a bit weird when you remove all shortcuts. We didn't resolve on iOS. I suppose the user has chosen this setup. Same as having every single menu item on too is a bit strange. We could enforce it being 3-5 items, but that seems against the spirit of it.

AjeshRPai committed:
* Fixes: Lint error

AjeshRPai committed:
* Fixes: the logic of showing more quick start focus point

- The quick start focus point show/hide was dependent on the active
quick link items, which is incorrect. Irregardless if the list item
 is in quick links or not, the quick start focus point should be shown
 on more quick link

AjeshRPai committed:
* Fixes: the logic of fetching the quick links for empty site

AjeshRPai committed:
* Fixes: the failing unit tests

AjeshRPai committed:
* Fixes: quick start check for get to know old quick start task - stats

zwarm committed:
[WIP] Temporarily turn off LoginTest so progress can continue

zwarm committed:
[WIP] Temporarily turn off SingupTest so progress can continue

zwarm committed:
Refactor: Adjust uiModel state to build if cards is null

zwarm committed:
Re-enable tests

zwarm committed:
Merge pull request #19282 from wordpress-mobile/issue/19270-quick-start-in-menu

Implements: the QuickStartFocus Point in more menu

zwarm committed:
Move components outside of MenuActivity class, but still in the same file

zwarm committed:
Refactor: adjust the customXMLWidgetView composable to be more efficient

zwarm committed:
Refactor: spacing adjustments

zwarm committed:
Implement layout direction

zwarm committed:
Refactor: use uiStringText where possible

zwarm committed:
Add state specifically for menu screen. Convert from MySiteCardAndItem

zwarm committed:
Refactor: swap MenuItemState for MySiteCardAndItem

zwarm committed:
Refactor: Re-order constructor parameters

zwarm committed:
Add initial set of tests

AjeshRPai committed:
Merge pull request #19289 from wordpress-mobile/issue/19270-menu-ui-cleanup

Unified Dashboard: Menu UI cleanup

AjeshRPai committed:
* Fixes: the base activity in PersonalizationActivity and MenuActivity

zwarm committed:
Add QuickStartMenuStep to support steps that are in the more menu.

zwarm committed:
Adjust tests to support QuickStartMenuStep

zwarm committed:
Refactor: use QuickStartMenuStep instead of activeTask because more focus is an interim step, not the active task

zwarm committed:
Refactor: have quickLins depend on quickStartMenuStep and not activeTask. More tap is an interim step, not the active task

zwarm committed:
Refactor tests to support QuickStartMenuStep for the more menu

zwarm committed:
Remove todo

zwarm committed:
Update tests to support QuickStartMenuStep

zwarm committed:
Refactor: Add quickStartEvent to OpenMore class

zwarm committed:
Refactor: Set quickStartEvent in OpenMore SiteNavigationAction

zwarm committed:
Refactor: Pass along quickStartEvent from OpenMore SiteNavigationAction to activityNavigator

zwarm committed:
Add intent KEY for quick start event

zwarm committed:
Set quickStartEvent, if available in openUnifiedSiteMenu

zwarm committed:
Refactor: add isFromMenu boolean to setActiveTask so the next step of the tutorial shows when on the more menu

zwarm committed:
Refactor: add separate click for more if in a quick start situation

zwarm committed:
Capture quickStartPrompts and snackbarMessage from quickStartRepo. Hanlde QuickStartEvent

zwarm committed:
Observe quickStartPrompts and snackbarMessage from quickStartRepo.Pass quickStartEvent to the viewmodel

zwarm committed:
Refactor: swap the double bang for let to handle resetting the menuStep

AjeshRPai committed:
+ Adds: the tracking for personalization of shortcuts

zwarm committed:
Refactor: set the constants to public so they can be used for quickstart snackbars within more menu

zwarm committed:
Add callback to tie in compose views; they don't show traditional snackbars, so a hook is needed to properly show.This is a lightweight implementation to support unified dashboard quick start.

zwarm committed:
Add support for showing quickStart snackbars for quick start.

zwarm committed:
Add support for showing quickStart snackbars for quick start by hooking into the composeSnackbarCallback in SnackbarSequencer

zwarm committed:
Add a delay before showing the snackbar because the QuickStartEvent arrives too quickly.

zwarm committed:
Refactor: set mocks for new di injections

AjeshRPai committed:
Merge pull request #19318 from wordpress-mobile/issue/18893-unified-dashboard-quick-start

Unified Dashboard: Quick Start Support

zwarm committed:
Merge branch 'trunk' into UI-Modernization-Remove-tabs-and-update-quick-links-layout

zwarm committed:
Add unit tests for backup and scan list items

zwarm committed:
Update tests to reflect the unified dashboard

zwarm committed:
Fix detekt issues

AjeshRPai committed:
+ Adds: the tracking for quick link clicks

AjeshRPai committed:
+ Adds: the tracking for menu viewmodel clicks

AjeshRPai committed:
* Fixes: Unit test

AjeshRPai committed:
* Fixes: Divider color between quick link items

AjeshRPai committed:
↑ Updates: the styling of menu items in more menu screen

AjeshRPai committed:
+ Adds: Spacing between title and text in personalization screen

AjeshRPai committed:
* Updates: the styling of shortcuts personalization screen

AjeshRPai committed:
Merge branch 'trunk' into UI-Modernization-Remove-tabs-and-update-quick-links-layout

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

AjeshRPai commented:
Hey @osullivanchris 
This PR is ready for another design review. 

> 1 - when I go into the full menu there's something wrong with the text styles. They have changed. They look heavier and grey rather than regular black text from before. And also the space between the icon and text looks tighter. The text style in the personalisation>shortcuts tab seems the same. It can also change to the way the menu was before.
> 
> <img alt="" width="250" src="https://user-images.githubusercontent.com/50576199/271560569-5192fa2b-0c9b-4e31-aa7c-f25e53976aad.png">

I have fixed this by making the font weight to normal and udpating the font color 


> 2 - the color of the separators between the shortcuts should be the same as the card border around it, but looks slightly different right now. I also removed all cards, then added others back, and the separator color changed to a darker color
> 
> <img alt="" width="250" src="https://user-images.githubusercontent.com/50576199/271560822-4b985c92-fac4-403e-97f0-aff2edc1fc12.png"> <img alt="" width="250" src="https://user-images.githubusercontent.com/50576199/271560849-f625b956-220e-454e-ad84-2bd748cdf89b.png">

Fixed this issue with commit https://github.com/wordpress-mobile/WordPress-Android/pull/19151/commits/85ac64b2eef3234ffaf3e4d63c9623f9a12d2ee6
 
> 3 - Can you add some extra space above the 'Active shortcuts' and 'Inactive shortcuts' headings? It all looks a bit jammed up together.
> 
> <img alt="" width="250" src="https://user-images.githubusercontent.com/50576199/271561167-a51374a5-1f0c-4400-ac2e-46a8970f5a7f.png">

Done 👍🏼 

> 4 - the style and label 'more' looks a bit weird when you remove all shortcuts. We didn't resolve on iOS. I suppose the user has chosen this setup. Same as having every single menu item on too is a bit strange. We could enforce it being 3-5 items, but that seems against the spirit of it.

I am not sure what we can do right now for fixing this TBH, I considered renaming the more to menu if there are no other quick links but that would add complexity to quick start snackbar messaging and all. As you said, if the user decides to go with no items or with all items I think we should let it be.

osullivanchris commented:
Thanks @AjeshRPai. Not at my machine or android phone now to install the build but looks good from the screenshots. No problem about the case with all shortcuts removed.

Re; the menu. The text looks better but still want to double check what is the padding between icon and text? Looks tight

AjeshRPai commented:
>Re; the menu. The text looks better but still want to double check what is the padding between icon and text? Looks tight

The padding now is 12 dp, earlier it was 8. I tried changing it to 16, felt its too much. I can make it to 16 dp if you think thats better.

osullivanchris commented:
I think go with 16 and then it's all good 👍🏻

AjeshRPai committed:
Merge branch 'trunk' into UI-Modernization-Remove-tabs-and-update-quick-links-layout

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt
#	WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt
#	WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt

AjeshRPai committed:
- Removes: Redundant ui model class for live data

AjeshRPai commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt:
```diff
@@ -969,12 +970,6 @@ class MySiteViewModel @Inject constructor(
         }
     }
 
-    data class UiModel(
```
Hey @zwarm 

I removed this class, the accountAvatarUrl and Account Name inside this class is not used at all. In no sites scenario, we are using the account details from `NoSites` State. Hence I thought we dont need this to wrap the state. WDYT

zwarm replied:
Thanks @AjeshRPai - that makes sense.

zwarm committed:
Add proper comment to ignored test

ravishanker commented:
@AjeshRPai - Please merge this first.  Fix any issues in subsequent PRs. Longer it waits, more conflicts, and merge issues it may face.  👍 :shipit:

AjeshRPai committed:
* Updates: the padding between icon and text in shortcuts and menu
